File: ios/Kinowo/Models/TopBarLayout.swift
#if canImport(CoreGraphics)
import CoreGraphics
#else
// `CoreGraphics` isn't on Linux, but Foundation re-exports `CGFloat`
// there — enough for this pure-logic type to build under `swift test` CI.
import Foundation
#endif

/// Decides, from the viewport width, where the search field lives.
///
/// On wide screens — iPads and landscape phones — there is room to sit
/// search on the top bar, after the "Wszystkie" date pill and before the
/// Filtry button. On narrow screens (portrait phones) there isn't, so it
/// stays as the floating pill at the bottom of the grid.
///
/// The threshold is shared verbatim with the Android app's `TopBarLayout`.
enum TopBarLayout {
    /// Viewport width (points) at or above which search moves inline. 600
    /// cleanly separates portrait phones (≤ ~440pt) from everything
    /// wider — landscape phones and iPads in either orientation.
    static let wideThreshold: CGFloat = 600

    static func searchInline(width: CGFloat) -> Bool { width >= wideThreshold }

    /// Whether all date pills should render at one uniform width.
    ///
    /// Equal width forces every pill to at least the widest label's intrinsic
    /// width (anything narrower clips that label), so a uniform row is only
    /// safe when the available width can hold `count` copies of the widest
    /// pill plus the `count - 1` gaps between them. When it can't, the pills
    /// fall back to their own intrinsic widths so every label still fits —
    /// fitting all the text takes priority over a uniform row.
    ///
    /// `intrinsicWidths` are the pills' natural (text + horizontal padding)
    /// widths; the caller measures them. Returns `false` for an empty set.
    static func datePillsEqualWidth(
        available: CGFloat,
        intrinsicWidths: [CGFloat],
        spacing: CGFloat
    ) -> Bool {
        guard let widest = intrinsicWidths.max() else { return false }
        let count = CGFloat(intrinsicWidths.count)
        return available >= count * widest + (count - 1) * spacing
    }
}

File: android/app/src/main/java/pl/kinowo/ui/TopBarLayout.kt
package pl.kinowo.ui

/**
 * Decides, from the viewport width, where the search field lives.
 *
 * On wide screens — tablets and landscape phones — there is room to sit
 * search on the top bar, after the "Wszystkie" date pill and before the
 * Filtry button. On narrow screens (portrait phones) there isn't, so it
 * stays as the floating pill at the bottom of the grid.
 *
 * The threshold is shared verbatim with the iOS app's `TopBarLayout`.
 */
object TopBarLayout {
    /** Width (dp) at or above which search moves inline. 600 cleanly
     *  separates portrait phones (≤ ~440dp) from everything wider —
     *  landscape phones and tablets in either orientation. */
    const val WideThresholdDp = 600

    fun searchInline(widthDp: Int): Boolean = widthDp >= WideThresholdDp

    /**
     * Decides whether a date pill stretches to share the row width equally
     * (via `Modifier.weight`) or keeps its intrinsic content width.
     *
     * On wide screens (landscape phones, tablets) there is ample room, so all
     * four pills — including "Wszystkie" — get equal weight and read as one
     * evenly-spaced segmented control.
     *
     * On narrow screens (portrait phones) the three short dated pills (Dziś /
     * Jutro / 7 dni) get weight and "Wszystkie" keeps its intrinsic width. The
     * dated pills therefore absorb ALL the width left between the 🎬 mark and the
     * Filtry button — the pill row fills the navbar with no trailing slack —
     * while "Wszystkie" stays exactly as wide as its long label needs. Their
     * compact-screen padding (see DatePill) is tuned so the filled share still
     * clears the labels at 14sp on a 360dp phone (Galaxy S24): they fill WITHOUT
     * clipping. Guarded by DayPillFitTest.
     */
    fun datePillFillsRow(isAnytime: Boolean, wide: Boolean): Boolean = wide || !isAnytime
}
